library/log.go

```diff
@@ -36,6 +42,47 @@ func (l *Log) AppendData(data []byte) {
 	l.SetData(append(l.GetData(), data...))
 }
 
+// MaskData reads through the log data and masks
+// all values provided in the string slice. If the
+// log is empty, we do nothing.
+func (l *Log) MaskData(secrets []string) {
+	// convert data to string
+	strData := string(l.GetData())
+	for _, secret := range secrets {
+		// escape regexp meta characters if they exist within value of secret
+		sanitizedStr := regexp.QuoteMeta(secret)
+
+		// find matches in logs for secrets surrounded by whitespace, start of string,
+		// quotation, colon, period, comma, and end of string.
+		// this prevents partial masking of secrets that contain a substring of another
+		// secret.
+		re := regexp.MustCompile((`(\s|^|"|:|'|\.|,)` + sanitizedStr + `(\s|$|"|:|'|\.|,)`))
+		matches := re.FindAllString(strData, -1)
+
+		// take all matches and mask them
+		for _, match := range matches {
+			// construct mask
+			mask := ""
+
+			// if secret was logged as the very first or last thing in the log, don't add
+			// boundary mask values.
+			if match[0] != secret[0] {
+				mask += string(match[0])
+			}
+			mask += constants.SecretLogMask
+			if match[len(match)-1] != secret[len(secret)-1] {
+				mask += string(match[len(match)-1])
+			}
+
+			// replace log with new mask
+			strData = strings.Replace(strData, match, mask, 1)
+		}
+	}
```

Comment:
To explain the complicated masking procedure, I am trying to address all the following scenarios:

Say your secret was the value `(vela**secret[xyz]{`

`QuoteMeta` will turn that value into `\(vela\*\*secret\[xyz\]\{` in order to find its occurrence in the logs. 

However, if we had another secret, say `xyz`, then there is a potential that the above secret would get masked to `(vela**secret[***]{`. This is why there needed to be the introduction of boundary characters, which are all the characters specified on line 59. It's not foolproof. For example, if we added the characters `[` and `]` to the mix, then the `xyz` secret would get skipped. Hopefully folks can chime in on how best to approach these complicated arrangements of secrets within logs. 

For now, it will simply address scenarios like these:

`"$ echo $SECRET\n(vela**secret[xyz]{\n` --> `"$ echo $SECRET\n***\n`
`(vela**secret[xyz]{\n/bin/sh: (vela**secret[xyz]{: not found` --> `***\n/bin/sh: ***: not found`

And so on. The various ways around the current state could be `<secretValue>` or `[secretValue]`, but that can be fixed by adding those to the pattern we're matching.